packages/xrpl/test/integration/transactions/ammBid.test.ts

```diff
@@ -0,0 +1,191 @@
+import { assert } from 'chai'
+import { AMMBid, AMMDeposit, AMMDepositFlags, IssuedCurrencyAmount } from 'xrpl'
+
+import { AMMInfoResponse, Wallet } from '../../../src'
+import serverUrl from '../serverUrl'
+import {
+  setupClient,
+  teardownClient,
+  type XrplIntegrationTestContext,
+} from '../setup'
+import { generateFundedWallet, setupAMMPool, testTransaction } from '../utils'
+
+describe('AMMBid', function () {
+  let testContext: XrplIntegrationTestContext
+  let wallet: Wallet
+  let wallet2: Wallet
+  let wallet3: Wallet
+  let currencyCode: string
+  let lptoken: IssuedCurrencyAmount
+
+  beforeAll(async () => {
+    testContext = await setupClient(serverUrl)
+    wallet = testContext.wallet
+    wallet2 = await generateFundedWallet(testContext.client)
+    wallet3 = await generateFundedWallet(testContext.client)
+    currencyCode = 'USD'
+
+    const ammInfoRes = await setupAMMPool(
```

Comment:
Should there be some sort of "reusable value" generation like we have in xrpl-py for this, so you don't need to generate an AMM for every single test? It would be useful for XLS-38d integration tests as well.

Reply:
Yeah that would definitely improve efficiency. But I'm trying to determine where the generated AMM would be saved to be queried by all tests. I'm thinking it can could be added to the test context`XrplIntegrationTestContext`.